# HG changeset patch
# User Mychaela Falconia <falcon@freecalypso.org>
# Date 1693510698 0
#      Thu Aug 31 19:38:18 2023 +0000
# Node ID a96cb97b66a221425071357950bfeb616c1ec9c7
# Parent  b515a97e5dff252b24c41fecc139f8db19f6a637
ringtools/imy: fix duplicate definition of tdma_durations[]

The bug was reported by Vadim Yanitskiy <fixeria@osmocom.org>,
although the present fix is slightly different from the contributed
patch: because main.c doesn't need this tdma_durations[] array
at all, let's simply remove the reference to this array from main.c
rather than turn it into an extern.

I no longer remember my original thought flow that resulted (by mistake)
in tdma_durations[] being multiply defined in main.c and durations.c.
My intent might have been to define all globals in main.c and have
the reference in durations.c be an extern - and I missed that extern -
but without clear memory, I have no certainty.  In any case, having
this data array defined in the same module that fills it (durations.c)
is sensible, so let's make it the new way.

diff -r b515a97e5dff -r a96cb97b66a2 ringtools/imy/main.c
--- a/ringtools/imy/main.c	Mon Aug 28 03:52:53 2023 +0000
+++ b/ringtools/imy/main.c	Thu Aug 31 19:38:18 2023 +0000
@@ -9,7 +9,6 @@
 char *imy_filename;
 char melody_str_buf[MELODY_BUF_SIZE];
 unsigned beats_per_min = 120;
-unsigned tdma_durations[6][4];
 FILE *outF;
 
 main(argc, argv)
